std/logger/core.d

```diff
@@ -1433,7 +1433,7 @@ logger by the user, the default logger's log level is LogLevel.info.
 
 Example:
 -------------
-sharedLog = new FileLogger(yourFile);
+sharedLog = cast(shared) new FileLogger(yourFile);
```

Comment:
should this not be `sharedLog = new shared(FileLogger)(yourFile);`?

Reply:
I would think so too, but that produces
```
Error: none of the overloads of `__ctor` are callable using a `shared` object
C:\D\dmd2\windows\bin64\..\..\src\phobos\std\logger\filelogger.d(40,5):        Candidates are: `std.logger.filelogger.FileLogger.this(const(string) fn, const(LogLevel) lv = LogLevel.all)`
C:\D\dmd2\windows\bin64\..\..\src\phobos\std\logger\filelogger.d(66,5):                        `std.logger.filelogger.FileLogger.this(const(string) fn, const(LogLevel) lv, Flag createFileNameFolder)`
C:\D\dmd2\windows\bin64\..\..\src\phobos\std\logger\filelogger.d(105,5):                        `std.logger.filelogger.FileLogger.this(File file, const(LogLevel) lv = LogLevel.all)`
```

I have found zero documentation on creating shared instances.

Reply:
Should there have been a `shared this()`?

Reply:
Oh, no shared constructor. That would also solve the problem.

Reply:
OK, that would be a better solution. The example should also be a unittest. I'll see if I can find time to correct this.

Reply:
IMO this PR is okay to merge, as I think going down the rabbit hole of shared ctor's is not worth the trouble.

Reply:
Thank you, but it looks like #8783 is a nicer fix. I'll see if I can revive that one, and adjust the docs accordingly.